### ISSUE TITLE
lightweight xml parsing

### ISSUE THREAD

**Issue by briceburg** [BlueAcornInc/Node-Magento-DB-Tools#1]
can we use a lightweight xml parsing library like xmldoc / sax versus libxmljs? libxmljs is adding quite bit of dependencies (python, gcc, make for node-gyp) and increases `npm install` time 10x.

**Comment by romiro** [comment]
@briceburg This was a consolation after working through a number of xml libraries, you can see it way back in the commit history where I get angry at the world of XML for having shit libs for node.

It's a possibility, but could require big changes. Is it still an issue now with getting it to work in alpine?

**Comment by romiro** [comment]
@briceburg  I could in theory try the work I ripped out during this commit: https://github.com/BlueAcornInc/Node-Magento-DB-Tools/commit/83f875652af98ff4591037b176eb1b507260e6a5

That is where I removed xmldom + xpath modules in favor of libxmljs

**Comment by briceburg** [comment]
This issue kind of killed me on alpine, and it does slow down the build process 10x on debian.Alpine port was dumped after further issues w/ sqlite. So we're OK *now*, although I recommend a lightweight solution going forward w/ the next iteration of this tool. Had to do the same thing w/ nodegit vs. git-promise to fix docker build times and unruly dependencies.

**Comment by romiro** [comment]
@briceburg But I mean, long compile times aren't unheard of in the world of programming, right? And in this specific case, the dbtools node is just going to be built _once_ through docker, barring migrations or meltdowns, correct? So would it be fair to say it's a negligible concern that's just irritating as hell right now in the "dev" process?

**Comment by briceburg** [comment]
@romiro yeah but that's always the case. lets just try to avoid compiling deps if possible going forward.
